src/Symfony/Bridge/Doctrine/ManagerRegistry.php

```diff
@@ -37,6 +39,35 @@ protected function getService($name)
      */
     protected function resetService($name)
     {
-        $this->container->set($name, null);
+        if (!$this->container->initialized($name)) {
+            return;
+        }
+        $manager = $this->container->get($name);
+
+        if (!$manager instanceof LazyLoadingInterface) {
+            @trigger_error(sprintf('Resetting a non-lazy manager service is deprecated since Symfony 3.2 and won\'t be supported in version 4.0. Set the "%s" service as lazy instead.', $name));
+
+            $this->container->set($name, null);
+
+            return;
+        }
+        $manager->setProxyInitializer(
+            function (&$wrappedInstance, LazyLoadingInterface $manager) use ($name) {
+                $this->container->set($name, null);
+                $proxy = $this->container->get($name);
+
+                if ($proxy instanceof LazyLoadingInterface && !$proxy->isProxyInitialized() && $proxy instanceof ValueHolderInterface) {
+                    $proxy->initializeProxy();
+                    $wrappedInstance = $proxy->getWrappedValueHolderValue();
+                } else {
+                    $wrappedInstance = $proxy;
+                }
+
+                $manager->setProxyInitializer(null);
+                $this->container->set($name, $manager);
```

Comment:
The deprecation is about private services. Just make follow deprecation notices and set the *em* service public.
The issue are separate (but we missed a test).